src/h5web/vis-packs/core/heatmap/HeatmapMesh.tsx

```diff
@@ -20,20 +22,37 @@ interface Props {
   values: number[];
   domain: Domain;
   colorMap: ColorMap;
-  scaleType: ScaleType;
+  scaleType: ScaleParams;
   invertColorMap?: boolean;
   alphaValues?: number[];
   alphaDomain?: Domain;
 }
 
 const CMAP_SIZE = 256;
 
-const SCALE_FUNC: Record<ScaleType, (val: number) => number> = {
-  [ScaleType.Linear]: (val) => val,
-  [ScaleType.Log]: Math.log10,
-  [ScaleType.SymLog]: (val) => Math.sign(val) * Math.log10(1 + Math.abs(val)),
-  [ScaleType.Sqrt]: Math.sqrt,
-};
+function computeScaledDomain(
+  scaleType: Omit<ScaleType, 'gamma'> | ScaleType.Gamma,
+  domain: Domain
+): Domain {
+  switch (scaleType) {
+    case ScaleType.Linear:
+    case ScaleType.Gamma:
+      return domain;
```

Comment:
> Shouldn't the gamma scale have this scale function here: `val => Math.pow(val, exponent)`, and in the shader: `return oneOverRange * (pow(value, gammaExponent) - min));`
> 
> It's equivalent,

It is not equivalent: the `pow(value, gammaExponent)` must be done when `value` is in `[0,1]` which is not the case with the formula you wrote

Reply:
The root of my comment is the fact that this `return domain;` is inconsistent with the rest. It shows that there's a lot of coupling between this function (and previously the `SCALE_FUNC` record) and the shader implementations for each scale. It might be clearer to do the necessary operations on the domain from inside the shader scale implementations

Reply:
Adding more operations into the shader is probably not what we want, so perhaps we could pass more uniforms regardless of whether they're actually needed? I don't know... Anyway, not hugely important.